entc/gen/graph.go

```diff
@@ -643,17 +644,25 @@ func (a assets) write() error {
 
 // format runs "goimports" on all assets.
 func (a assets) format() error {
-	for _, file := range a.files {
-		path := file.path
-		src, err := imports.Process(path, file.content, nil)
-		if err != nil {
-			return fmt.Errorf("format file %s: %v", path, err)
-		}
-		if err := ioutil.WriteFile(path, src, 0644); err != nil {
-			return fmt.Errorf("write file %s: %v", path, err)
-		}
+	var eg errgroup.Group
+
+	for i := range a.files {
+		idx := i
+		f := a.files[idx]
```

Comment:
it's a bit odd to declare this outside the func below.

Reply:
Has to happen to support errorgroup, the index has to be redeclared each iteration outside the goroutine. Same for the file element. It's just because it's not a `go func(f file){}(a.files[i])`

Reply:
But yes, it can be shortened to `f := a.files[i]`